lib/ui/hooks.dart

```diff
@@ -65,30 +66,31 @@ String/*?*/ _localeClosure() {
 
 @pragma('vm:entry-point')
 // ignore: unused_element
-_LocaleClosure/*?*/ _getLocaleClosure() => _localeClosure;
+_LocaleClosure? _getLocaleClosure() => _localeClosure;
 
 @pragma('vm:entry-point')
 // ignore: unused_element
-void _updateLocales(List<String/*!*/>/*!*/ locales) {
+void _updateLocales(List<String> locales) {
   const int stringsPerLocale = 4;
   final int numLocales = locales.length ~/ stringsPerLocale;
-  window._locales = List<Locale>(numLocales);
+  final List<Locale> newLocales = <Locale>[];
```

Comment:
We've discussed it, I can raise it again.  For this example, you could consider using a list comprehension.  I don't know how well optimized those are yet though.  @a-siva do you know if code like:
```
  var l = [
    for(int i = 0;i<count;i++) {
      i
  }];
```
will get compiled to pre-allocate the list?

Reply:
Also consider List.generate.

Reply:
This is a (relative) lot of computation for creating each element. Unless `numLocales` is big, or this code is called in a tight loop, I doubt the list growing is going to be visible.

The memory overhead of a growable list over a fixed-length list is real, but again, unless this gets called very often, it's probably not a problem.

I'd go for either:
```dart
  String? nullIfEmpty(String s) => s.isNotEmpty ? s : null;
  var newLocales = <Locale>[for (var i = 0; i < locales.length; i += stringsPerLocale) 
     Locale.fromSubtags(
         languageCode: locales[i],
         countryCode: _nullIfEmpty(locales[i + 1]), 
         scriptCode: _nullifEmpty(locales[i + 2]),
     )];
```
or `List.generate` if you really, really want a fixed-length list.

Either would be more readable than repeated addition.

(It would be nice to have `let` in list comprehensions, though!)